### PR TITLE
Fetching approval the good 'ol way

### PR DESCRIPTION
**Pull request by Alexintosh**
##### Checklist

- [ ] Linter status: 100% pass
- [ ] Changes don't break existing behavior
- [ ] Test coverage hasn't decreased

### CODE REVIEW

File: src/stores/eth.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ import { subject } from './eth/observables.js';
 import { trackEthBalance } from './eth/lifecycle.js';
 
 export { allowances, balances, eth, pools } from './eth/writables.js';
-export { balanceKey, functionKey } from './eth/keys.js';
+export { balanceKey, functionKey, getAllowanceKey } from './eth/keys.js';
```
Comment (jordaniza) [inline src/stores/eth.js:20]: Imported but unused, flagging

**Review by gbonumore — APPROVED** [review]

File: src/pages/DoughStaking.svelte [hunk 2]
```diff
@@ -296,9 +303,16 @@
                     }
                   }, 1000);
 
-                  approveToken($eth)
+                  let shouldResetAllowance = false;
+                  if(stakeAmount.bn.isGreaterThan($allowances[allowanceKey]) && $allowances[allowanceKey] > 0) {
```
Comment (jordaniza) [inline src/pages/DoughStaking.svelte]: Condition 1 is BN comparison 
Condition 2 is a `number` comparison

Just want to flag this as well. Checking that the right operators are being used here, i know you can pass a number as a comparison to a BN, but then I'm wondering if value of allowances[allowanceKey] is applicable in both circumstances.

File: src/helpers/staking.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ import { get } from 'svelte/store';
 export let sharesTimeLock = false;
 export let veDOUGH = false;
 export let merkleTreeDistributor = false;
-export const minLockAmount = 1;
+export const minLockAmount = 0;
```
Comment (dantop114) [inline src/helpers/staking.js:27]: any reason this was 1?

File: src/pages/DoughStaking.svelte (hunk 2 above)
Comment (Alexintosh) [reply src/pages/DoughStaking.svelte]: good spotted, changing it.

Comment (dantop114) [reply src/pages/DoughStaking.svelte]: +1

File: src/stores/eth.js (hunk 1 above)
Comment (Alexintosh) [reply src/stores/eth.js:20]: in this case it's shadowing exporting the method from the eth file.

**Review by jordaniza — APPROVED** [review]

File: src/helpers/staking.js (hunk 3 above)
Comment (Alexintosh) [reply src/helpers/staking.js:27]: Not that I'm aware of.

**Pull request merged**